HackRF_Settings.cpp

```diff
@@ -251,10 +251,17 @@ std::vector<std::string> SoapyHackRF::listGains( const int direction, const size
 	std::vector<std::string> options;
 	if ( direction == SOAPY_SDR_RX )
 	{
-		options.push_back( "LNA" );
+	// in gr-osmosdr/lib/soapy/ soapy_sink_c.cc and soapy_source_c.cc expect if_gain at front and bb_gain at back
+		options.push_back( "LNA" );						// RX: if_gain
+		options.push_back( "AMP" );						// RX: rf_gain
+		options.push_back( "VGA" );						// RX: bb_gain
+	}
+	else
+	{
+		options.push_back( "VGA" );						// TX: if_gain
+		options.push_back( "AMP" );						// TX: rf_gain
+		options.push_back( "LNA" );						// TX: bb_gain (not used on tx)
```

Comment:
You should leave LNA gain out of TX if its not there. Otherwise getGainRange() and setGain(overall...) will probably fail, SoapyUtil --probe too.

I'm not sure what the consequences of that are or if its a problem. Perhaps if gain will remain VGA and bb gain will now be AMP?

Reply:
I mostly agree with your suggestion, and I will make a commit to reflect that. This corrects the SoapySDRUtil -- probe issue (good catch). 

And yes, if someone uses the gr-osmosdr Sink block, with the SoapyHackRF driver, and puts a value in the BB_Gain setting, it will change the AMP gain. I don't see any way around that.